### ISSUE TITLE
WebAssemblyRuntime.InvokeJS get value from callback?

### ISSUE THREAD

**Issue by Ali-YousefiTelori** [nventive/Uno#65]
Hi,
How can I get callback result from Uno.Foundation.WebAssemblyRuntime.InvokeJS ?
for example like Promise in javascript etc...

**Comment by carldebilly** [comment]
You need to call back your code from js/ts. We're already doing that for the wasm http handler if you want an example.

**Comment by Ali-YousefiTelori** [comment]
Thank you, I don't know ts, but i make this example:
C#
```csharp
            var data = Uno.Foundation.WebAssemblyRuntime.InvokeJS(@"openFileDialog();
            Console.WriteLine("result: " + data);
```

and js:
```
async function openFileDialog() {
    function resolveAfter2Seconds(x) {
        return new Promise(resolve => {
            setTimeout(() => {
                resolve(x);
            }, 2000);
        });
    }

    var x = await resolveAfter2Seconds(10);
    return x;
}
```

this always show me result "result: [object Promise]" not real data.

**Comment by jeromelaban** [comment]
Async functions are not supported by InvokeJS, which is synchronous. You need to call a C# method manually at [the end like this](https://github.com/nventive/Uno/blob/8c71c7bcc24114d2f621fab0fc4432289ffaf302/src/Uno.UI.Wasm/ts/HttpClient.ts#L72).

**Comment by carldebilly** [comment]
@Ali-YousefiTelori @MatFillion
This feature has been merged into `master` and available as version `3.0.0-dev.25` on [nuget](https://www.nuget.org/packages/Uno.UI/3.0.0-dev.25).
